docs/docs/low_level_app.md

```diff
@@ -0,0 +1,159 @@
+
+```js
+const contractName = "Donations";
+const DonationsV0 = Contracts.getFromLocal('DonationsV0')
+await app.setImplementation(DonationsV0, contractName)
+const donationsV0 = await app.createProxy(DonationsV0, contractName, 'initialize', [owner])
+```
+
+Remember that the proxy is the contract that will receive the calls and hold the storage, while delegating its behavior to the implementation contract, enabling us to upgrade it.
+
+### 4. Link a standard library
+
+Now let's suppose we want to give some sort of retribution to the donors, so we mint new [ERC721](http://erc721.org/) cryptocollectibles for each donation. 
+
+For this, we link the [ZeppelinOS standard library](https://www.npmjs.com/package/openzeppelin-zos) to our application by running:
+
+```sh
+npm install openzeppelin-zos --save
+```
+
+Done! Now we can easily mint non-fungible tokens from our smart contract:
+
+```sol
+pragma solidity ^0.4.21;
+
+import "./DonationsV0.sol";
+import "openzeppelin-zos/contracts/token/ERC721/MintableERC721Token.sol";
+
+contract DonationsV1 is DonationsV0 {
+  using SafeMath for uint256;
+
+  MintableERC721Token public token;
+  uint256 public numEmittedTokens;
+
+  function setToken(MintableERC721Token _token) external {
+    require(_token != address(0));
+    require(token == address(0));
+    token = _token;
+  }
+
+  function donate() payable public {
+    super.donate();
+    token.mint(msg.sender, numEmittedTokens);
+    numEmittedTokens = numEmittedTokens.add(1);
+  }
+}
+```
+
+Notice that by doing this, our contract will interact directly with the on-chain ZeppelinOS standard library, so there is no need to deploy nor maintain the `MintableERC721Token` contract ourselves, as we rely on the ZeppelinOS network.
+
+### 5. Upgrade to the new version
+
+To upgrade our app, we need to create a new version and reference the ZeppelinOS standard library release
+
+```js
+const stdlibAddress = "0xA739d10Cc20211B973dEE09DB8F0D75736E2D817" // ropsten release
+const newVersion = '0.0.2'
+await app.newVersion(newVersion, await getStdLib(stlibAddress))
+```
+
+Next, we register the new implementation of our donations contract
+
+```js
+const DonationsV1 = Contracts.getFromLocal('DonationsV1')
+await app.setImplementation(DonationsV1, contractName)
+```
+
+And upgrade the proxy
+
+```js
+await app.upgradeProxy(donationsV0.address, null, contractName)
+// We wrap the same old proxy address with the new interface
+donationsV1 = DonationsV1.at(donationsV0.address)
+```
+
+Then we create a proxy to the standard library version of the ERC721 contract, declaring our `donationV1` proxy address as the owner
+
+```js
+const token = await app.createProxy(
+  MintableERC721Token, 
+  tokenClass,
+  'initialize',
+  [donationsV1.address, tokenName, tokenSymbol]
+)
+```
+
+Finally, we set it to our upgradeable contract
```

Comment:
set it to our upgradeable contract -> set it as the token of our upgradeable contract